samza-test/src/test/java/org/apache/samza/test/processor/TestStreamApplication.java

```diff
@@ -141,7 +142,7 @@ public String toString() {
     }
 
     static TestKafkaEvent fromString(String message) {
-      String[] messageComponents = message.split("|");
+      String[] messageComponents = StringUtils.split(message, "|");
```

Comment:
This was previously wrong. Calling `split` on the string directly expects a regex string and "|" is a regex character that needed to be escaped. Switched to `StringUtils.split` as it does what we thought was intended.

Reply:
Thanks for catching this!